src/mame/drivers/a2600.cpp

```diff
@@ -502,12 +685,136 @@ void a2600_state::a2600p(machine_config &config)
 }
 
 
-ROM_START( a2600 )
-	ROM_REGION( 0x2000, "maincpu", ROMREGION_ERASEFF )
+void a2600_pop_state::a2600_pop(machine_config &config)
+{
+	/* basic machine hardware */
+	M6507(config, m_maincpu, MASTER_CLOCK_NTSC / 3);
+	m_maincpu->set_addrmap(AS_PROGRAM, &a2600_pop_state::memory_map);
+
+	/* video hardware */
+	TIA_NTSC_VIDEO(config, m_tia, 0, "tia");
+	m_tia->read_input_port_callback().set(FUNC(a2600_pop_state::a2600_read_input_port));
+	m_tia->databus_contents_callback().set(FUNC(a2600_pop_state::a2600_get_databus_contents));
+	m_tia->vsync_callback().set(FUNC(a2600_pop_state::a2600_tia_vsync_callback));
+
+	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
+	m_screen->set_raw(MASTER_CLOCK_NTSC, 228, 26, 26 + 160 + 16, 262, 24 , 24 + 192 + 31);
+	m_screen->set_screen_update("tia_video", FUNC(tia_video_device::screen_update));
+
+	/* sound hardware */
+	SPEAKER(config, "mono").front_center();
+	TIA(config, "tia", MASTER_CLOCK_NTSC/114).add_route(ALL_OUTPUTS, "mono", 0.90);
+
+	/* devices */
+#if USE_NEW_RIOT
+	MOS6532_NEW(config, m_riot, MASTER_CLOCK_NTSC / 3);
+	m_riot->pa_rd_callback().set(FUNC(a2600_pop_state::switch_A_r));
+	m_riot->pa_wr_callback().set(FUNC(a2600_pop_state::switch_A_w));
+	m_riot->pb_rd_callback().set_ioport("SWB");
+	m_riot->pb_wr_callback().set(FUNC(a2600_pop_state::switch_B_w));
+	m_riot->irq_wr_callback().set(FUNC(a2600_pop_state::irq_callback));
+#else
+	RIOT6532(config, m_riot, MASTER_CLOCK_NTSC / 3);
+	m_riot->in_pa_callback().set(FUNC(a2600_pop_state::switch_A_r));
+	m_riot->out_pa_callback().set(FUNC(a2600_pop_state::switch_A_w));
+	m_riot->in_pb_callback().set_ioport("SWB");
+	m_riot->out_pb_callback().set(FUNC(a2600_pop_state::switch_B_w));
+	m_riot->irq_callback().set(FUNC(a2600_pop_state::irq_callback));
+#endif
+
+	VCS_CONTROL_PORT(config, CONTROL1_TAG, vcs_control_port_devices, "joy");
+	VCS_CONTROL_PORT(config, CONTROL2_TAG, vcs_control_port_devices, "joy");
+
+	ADDRESS_MAP_BANK(config, "bank").set_map(&a2600_pop_state::banked_map).set_options(ENDIANNESS_LITTLE, 8, 22, 0x1000);
```

Comment:
Is there really nothing in common with the home console that can be brought out to a common function?

Reply:
Moved the common parts to a2600_base_ntsc